tools/src/wyvern/target/corewyvernIL/decl/DefDeclaration.java

```diff
@@ -33,6 +34,15 @@ public DefDeclaration(String methodName, List<FormalArg> formalArgs,
 	}
 
 	@Override
+	public boolean containsResource() {
+		return this.hasResource;
+	}
+
+	public void setHasResource(boolean hasResource) {
```

Comment:
This should be private, as we don't want the resource attribute to be set externally.